index.js

```diff
@@ -21,9 +21,15 @@ var date_with_offset = /^\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d(\.\d\d\d)? (Z|(-|\+|)
 var date_rfc_2822_regex = /^\d\d-\w\w\w-\d\d\d\d \d\d:\d\d:\d\d (\+|-)\d\d\d\d$/;
 var local_date_regex = /^\d\d\d\d-\d\d-\d\d[T ]\d\d:\d\d(:\d\d(\.\d\d\d)?)?$/;
 
+var currentTime = null;
+
 function MockDate(param) {
   if (arguments.length === 0) {
-    this.d = new _Date();
+    if (typeof currentTime === 'number') {
+        this.d = new _Date(currentTime);
```

Comment:
In reality, the current time is not constant, and code often has subtle bugs resulting from assuming two nearby `Date.now()` calls return the same value... would `new _Date(currentTime++)` be a better choice (with README saying `starting from a given timestamp`)?

Reply:
@Jimbly What I actually need is what cypress does with clock (but using that feature there is no timezone simulation): https://docs.cypress.io/api/commands/clock#Examples
So what do you think about having a `tick` function with a value for increasing the time?
The problem with `currentTime++` is that it's not controlled, and in a test it's important to know what to expect. It's hard to follow `currentTime++` so an assert would be impossible to be made.

Reply:
An assert could be made, but it, like any safe code dealing with elapsed time, might need to be checking a range instead of an exact value ^_^.  Understandably possibly harder to follow though.  A tick function is not unreasonable, however, unlike that Cypress example, it's not going to affect things like setInterval, so maybe a little confusing.  If you want to use Cypress, and mock timezones, you should just be able to use them both - as long Cypress mocks the `Date` object _before_ you require/import `timezone-mock`, then `timezone-mock` will use Cypress's mocked Date, and they should work together.  A simple change to timezone-mocks's `register()` to move the `_Date` initialization to there (`if (Date !== MockDate) { _Date = Date; }`) would make that even easier (just the order of registering/mocking would matter, not importing).

Reply:
That unfortunately doesn't work out of the box, as `cy.clock()` changes the Date object too late, when this library already saved a reference to the `Date` object.
In order to work, and also to give a bit of a freedom with combining other Date manipulation libraries would be to save the reference to the `Date` object in this library on `register` (`_Date = Date`).
I think this change would give much more flexibility. What do you think?